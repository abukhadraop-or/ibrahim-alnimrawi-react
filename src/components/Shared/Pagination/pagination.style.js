import styled from "styled-components";
export const PaginationContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  cursor: default;
  margin: 0 2rem;
`;
export const PaginateItem = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  color: #5cb85c;
  border: 1px solid #ddd;
  width: 2rem;
  height: 2rem;
  &:hover {
    background-color: #ddd;
  }
  &:active {
    border: 1px solid #000;
  }
`;
export const ActiveItem = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #5cb85c;
  color: #fff;
  border: 1px solid #ddd;
  width: 2rem;
  height: 2rem;
`;
