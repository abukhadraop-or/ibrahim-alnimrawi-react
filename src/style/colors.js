const colors = {
  black: '#000000',
  darkCyan_limeGreen: '#09733a',
  darkGray: '#999999',
  darkGrayishBlue: '#7d858e',
  gray: '#bbbbbb',
  lightGrayishBlue: '#b8c3e0',
  Mercury: '#E5E5E5',
  moderateLimeGreen: '#5cb85c',
  veryDarkGrayishBlue: '#373a3c',
  veryDarkGrayishBlue2: '#404b59',
  veryDarkGrayishBlue3: '#2F3236',
  veryLightGray_mostlyWhite: '#f3f3f3',
  veryLightGray: '#dddddd',
  white: '#ffffff',
};
export default colors;
