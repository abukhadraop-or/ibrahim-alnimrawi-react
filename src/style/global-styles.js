import { createGlobalStyle } from 'styled-components';

/* istanbul ignore next */
const GlobalStyle = createGlobalStyle`
  body {
    font-family: Serif;
    *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
  padding: 0;
  margin: 0;
  }
`;

export default GlobalStyle;
