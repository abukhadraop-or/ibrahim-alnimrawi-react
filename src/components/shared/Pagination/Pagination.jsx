import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { PaginateItem, PaginationContainer } from './pagination.style';

/**
 * Render a <Pagination> component
 *
 * @param {Object}   props                 The props object.
 * @param {Number}   props.pageSize        The number of articles per page.
 * @param {Number}   props.currentPage     The current page number.
 * @param {Function} props.onPageChange    Function to handle page change.
 * @param {Number}   props.itemCount       The number of articles.
 *
 * @return {JSX.Element}
 */
const Pagination = ({ pageSize, currentPage, onPageChange, itemsCount }) => {
  const numberOfPages = Math.ceil(itemsCount / pageSize);
  const pages = _.range(1, numberOfPages + 1);

  if (numberOfPages === 1 || numberOfPages === 0) {
    return null;
  }

  return (
    <PaginationContainer>
      {pages.map((page) => (
        <PaginateItem
          active={page === currentPage}
          onClick={() => onPageChange(page)}
          key={page}
        >
          {page}
        </PaginateItem>
      ))}
    </PaginationContainer>
  );
};

export default Pagination;

Pagination.propTypes = {
  pageSize: PropTypes.number,
  currentPage: PropTypes.number,
  onPageChange: PropTypes.func,
  itemsCount: PropTypes.number,
};

Pagination.defaultProps = {
  pageSize: 10,
  currentPage: 1,
  onPageChange: null,
  itemsCount: 0,
};
