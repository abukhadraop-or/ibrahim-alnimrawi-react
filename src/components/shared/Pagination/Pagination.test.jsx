import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Pagination from 'components/shared/Pagination/Pagination';
import Enzyme, { shallow } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { PaginateItem } from './pagination.style';

Enzyme.configure({ adapter: new Adapter() });

describe('<pagination />', () => {
  it('should be defined', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<Pagination />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be zero', () => {
    const wrapper = shallow(
      <Pagination
        pageSize={10}
        currentPage={1}
        onPageChange={jest.fn()}
        itemsCount={5}
      />
    );
    expect(wrapper.find(PaginateItem).length).toBe(0);
  });

   
  it('should be 3', () => {
    const wrapper = shallow(
      <Pagination
        pageSize={10}
        currentPage={1}
        onPageChange={jest.fn()}
        itemsCount={23}
      />
    );
    expect(wrapper.find(PaginateItem).length).toBe(3);
  });
  it('call onClick function ', () => {
    const onPageChange = jest.fn();
    const wrapper = shallow(
      <Pagination
        pageSize={10}
        currentPage={1}
        onPageChange={onPageChange}
        itemsCount={23}
      />
    );
    const pItem = wrapper.find(PaginateItem);
    pItem.at(1).simulate('click');

    expect(onPageChange).toHaveBeenCalledWith(2);
  });
});
