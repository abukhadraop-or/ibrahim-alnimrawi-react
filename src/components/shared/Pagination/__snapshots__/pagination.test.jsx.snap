// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<pagination /> should match the snapshot 1`] = `null`;
